### PR TITLE
Add low-resolution and high-resolution scaling tests 

### PR DESCRIPTION
**Pull request by sriharshakandala**
Add low-resolution and high-resolution scaling tests and compare with Tempest.

### CODE REVIEW

File: .buildkite/scaling/pipeline.yml [hunk 1]
```diff
@@ -67,96 +67,184 @@ steps:
...
+        config: cpu
+        queue: central
+        slurm_nodes: 1
+        slurm_tasks_per_node: 2
+        slurm_mem: 64GB
 
-    - label: ":computer: MPI Held-Suarez scaling test(ρe_tot) - (64) processes"
-      command: "mpiexec julia --color=yes --project=examples examples/hybrid/driver.jl --job_id sphere_held_suarez_rhoe --forcing held_suarez --enable_threading false --upwinding none --t_end 1hours --dt 100secs --z_elem 40 --h_elem 16 --kappa_4 1e15" 
-      artifact_paths: "sphere_held_suarez_rhoe/scaling_data_64_processes.jld2"
+    - label: ":computer: MPI Held-Suarez high resolution scaling test(ρe) - (3) processes"
```
Comment (szy21) [inline .buildkite/scaling/pipeline.yml:158]:
```suggestion
    - label: ":computer: MPI Held-Suarez high resolution scaling test(ρe_tot) - (3) processes"
```

File: .buildkite/scaling/pipeline.yml [hunk 2]
```diff
@@ -67,96 +67,184 @@ steps:
...
+        CLIMACORE_DISTRIBUTED: "MPI"
+        NPROCS: 3
+      agents:
+        config: cpu
+        queue: central
+        slurm_nodes: 1
+        slurm_tasks_per_node: 3
+        slurm_mem: 64GB
+
+    - label: ":computer: MPI Held-Suarez high resolution scaling test(ρe) - (6) processes"
```
Comment (szy21) [inline .buildkite/scaling/pipeline.yml:175]:
```suggestion
    - label: ":computer: MPI Held-Suarez high resolution scaling test(ρe_tot) - (6) processes"
```

File: .buildkite/scaling/pipeline.yml [hunk 3]
```diff
@@ -67,96 +67,184 @@ steps:
...
         CLIMACORE_DISTRIBUTED: "MPI"
+        NPROCS: 6
+      agents:
+        config: cpu
+        queue: central
+        slurm_nodes: 1
+        slurm_tasks_per_node: 6
+        slurm_mem: 64GB
+
+    - label: ":computer: MPI Held-Suarez high resolution scaling test(ρe) - (24) processes"
```
Comment (szy21) [inline .buildkite/scaling/pipeline.yml:192]:
```suggestion
    - label: ":computer: MPI Held-Suarez high resolution scaling test(ρe_tot) - (24) processes"
```

File: .buildkite/scaling/pipeline.yml [hunk 4]
```diff
@@ -67,96 +67,184 @@ steps:
...
+        CLIMACORE_DISTRIBUTED: "MPI"
+        NPROCS: 24
+      agents:
+        config: cpu
+        queue: central
+        slurm_nodes: 1
+        slurm_tasks_per_node: 24
+        slurm_mem: 128GB
+
+    - label: ":computer: MPI Held-Suarez high resolution scaling test(ρe) - (54) processes"
```
Comment (szy21) [inline .buildkite/scaling/pipeline.yml:209]:
```suggestion
    - label: ":computer: MPI Held-Suarez high resolution scaling test(ρe_tot) - (54) processes"
```

**Review by szy21 — APPROVED** [review]
LGTM, thanks! For the high-resolution scaling plot, I wonder if it's better to plot the full tempest result, rather than cutting off at nproc=54?

**Comment by sriharshakandala** [comment]
> LGTM, thanks! For the high-resolution scaling plot, I wonder if it's better to plot the full tempest result, rather than cutting off at nproc=54?

I am restricting to the data we are comparing for the moment. Larger runs, requiring multiple nodes, have significant wait time to launch. We can add them in the near future!

**Pull request merged**